Feature 1001 blocking statanalysis

CPKalb opened the pull request:
## Pull Request Testing ##

- [ ] Describe testing already performed for these changes:</br>
verified that the runs call stat_analysis and produce reasonable output

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
run the blocking and weather regime use cases to verify they work (UserScript_fcstGFS_obsERA_Blocking.conf and UserScript_fcstGFS_obsERA_WeatherRegime.conf).  Both should end with calls to stat_analysis

- [ ] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes or No]**
Yes

- [ ] Do these changes include sufficient testing updates? **[Yes or No]**
Yes

- [ ] Will this PR result in changes to the test suite? **[Yes or No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>
Yes

- [ ] Please complete this pull request review by **[Fill in date]**.</br>
-before the next release

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

CPKalb committed:
Added stat_analysis calls to the blocking and weather regime use cases

CPKalb committed:
Fixed documentation error

georgemccabe committed:
Merge branch 'develop' into feature_1001_blocking_statanalysis

georgemccabe reviewed (changes requested):
I reviewed the differences in the output for the 3 use cases that were modified.

UserScript_fcstGFS_obsERA_Blocking
* New mpr/IBL output stat files
* New images

UserScript_obsERA_obsOnly_Blocking
* Lines in images slightly shifted

UserScript_obsERA_obsOnly_WeatherRegime
* output directory name changed from UserScript_obsERA_obsOnly_WeatherRegime to UserScript_fcst_GFS_obsERA_WeatherRegime

The pull request notes mentioned that there will be differences in the output, but didn't specify what the differences should be so I assume that all of these changes are expected.

I merged the latest changes from the develop branch because this branch was behind by 23 commits.

It looks like there is now a new use case UserScript_fcst_GFS_obsERA_WeatherRegime that is not being run in the tests. The output directory in UserScript_obsERA_obsOnly_WeatherRegime seems to match this new use case name. Are both of these use cases needed? If not, please remove one of them and add the other to the tests.

Also, there are 2 blocking driver use cases -- one with just obs and one with fcst and obs. Do we need both of those? If the obs only use case is just a subset of the other use case, then I think it could be removed.

CPKalb commented:
The obs only blocking use case was originally created to duplicate Doug's
results and is over a different time period than the case which contains
both model and observations.  We could delete it in the repository, but I
would like to keep it for my records so I'd ask that we wait until I can
make a backup copy to do this.  It's interesting that the lines appear
slightly changed in the   UserScript_obsERA_obsOnly_Blocking use case as I
believe they come out almost identical for me.

For now I would like to keep both weather regime use cases.  I'm still in
conversation with Doug about the changes in the kmeans section (though I'm
having trouble duplicating the old environment).  I need the obs only use
case as that will be easier for him to evaluate changes since it matches
his original results.  We can delete the obs only weather regime once I get
this worked out with Doug.  I'll have to add the one to the tests.  I
thought it would get added automatically since it's new but it seems it did
not.


On Tue, Aug 31, 2021 at 8:30 AM George McCabe ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> I reviewed the differences in the output for the 3 use cases that were
> modified.
>
> UserScript_fcstGFS_obsERA_Blocking
>
>    - New mpr/IBL output stat files
>    - New images
>
> UserScript_obsERA_obsOnly_Blocking
>
>    - Lines in images slightly shifted
>
> UserScript_obsERA_obsOnly_WeatherRegime
>
>    - output directory name changed from
>    UserScript_obsERA_obsOnly_WeatherRegime to
>    UserScript_fcst_GFS_obsERA_WeatherRegime
>
> The pull request notes mentioned that there will be differences in the
> output, but didn't specify what the differences should be so I assume that
> all of these changes are expected.
>
> I merged the latest changes from the develop branch because this branch
> was behind by 23 commits.
>
> It looks like there is now a new use case
> UserScript_fcst_GFS_obsERA_WeatherRegime that is not being run in the
> tests. The output directory in UserScript_obsERA_obsOnly_WeatherRegime
> seems to match this new use case name. Are both of these use cases needed?
> If not, please remove one of them and add the other to the tests.
>
> Also, there are 2 blocking driver use cases -- one with just obs and one
> with fcst and obs. Do we need both of those? If the obs only use case is
> just a subset of the other use case, then I think it could be removed.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplus/pull/1125#pullrequestreview-742808400>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFJHD4XF45T4WFZARMXL27LT7TRPPANCNFSM5DDKROYQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
*************************************************
Christina Kalb
Associate Scientist
National Center for Atmospheric Research
3450 Mitchell Lane
Boulder, CO 80301
***@***.***
303-497-2819
*************************************************

My working day may not be your working day. Please do not feel obliged to
reply to this email outside of your normal working hours.

CPKalb committed:
Updated some file names

CPKalb committed:
Merge branch 'feature_1001_blocking_statanalysis' of https://github.com/dtcenter/METplus into feature_1001_blocking_statanalysis

CPKalb committed:
Updated all_use_cases.txt

CPKalb commented:
Where is the second file to edit?  I found the all_use_cases.txt but I'm having trouble finding the second and I don't see it here (https://metplus.readthedocs.io/en/develop/Contributors_Guide/testing.html), but maybe I'm not looking in the right place in the documentation

georgemccabe commented:
> 
> 
> Where is the second file to edit? I found the all_use_cases.txt but I'm having trouble finding the second and I don't see it here ([metplus.readthedocs.io/en/develop/Contributors_Guide/testing.html](https://metplus.readthedocs.io/en/develop/Contributors_Guide/testing.html)), but maybe I'm not looking in the right place in the documentation

It is in the Adding Use Cases chapter:
https://metplus.readthedocs.io/en/develop/Contributors_Guide/add_use_case.html#add-new-category-to-test-runs

CPKalb committed:
Updated use case groups.json

CPKalb commented:
It looks like there are some differences in the blocking and weather regime analysis data.  These differences are okay and expected (I changed the model dataset for the Blocking and fixed a small bug).  How would I set the new images as truth?

georgemccabe commented:
> 
> 
> It looks like there are some differences in the blocking and weather regime analysis data. These differences are okay and expected (I changed the model dataset for the Blocking and fixed a small bug). How would I set the new images as truth?

There is a section in the Contributor's Guide Adding Use Cases chapter to update the truth data. However, I need to add instructions that will prevent conflicts between develop and develop-ref that cause issues. I can take care of this step for this PR. 

Here is the section for future reference:
https://metplus.readthedocs.io/en/develop/Contributors_Guide/add_use_case.html#create-a-pull-request-from-develop-into-develop-ref
